aten/src/ATen/Error.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <ATen/ATenGeneral.h> // for AT_API
+
+#include <cstdint>
+#include <cstdio>
+#include <stdexcept>
+#include <string>
+
+namespace at {
+namespace detail {
+struct SourceLocation {
+  const char* function;
+  const char* file;
+  uint32_t line;
+};
+} // namespace detail
+
+template <typename... FormatArgs>
+[[noreturn]] AT_API void error(
+    detail::SourceLocation source_location,
+    const char* format_string,
+    FormatArgs&&... format_args) {
+  static const size_t kMaximumErrorMessageLength = 4096;
```

Comment:
Let's make it `constexpr`. Otherwise it might trigger some VLA extensions when defining `buffer`